src/Azure.Functions.Cli/Actions/AzureActions/PublishFunctionAppAction.cs

```diff
@@ -29,7 +29,7 @@ internal class PublishFunctionAppAction : BaseFunctionAppAction
 
         private readonly ISettings _settings;
         private readonly ISecretsManager _secretsManager;
-        private const string _requiredNetFrameworkVersion = "6.0";
+        private static string _requiredNetFrameworkVersion = "6.0";
```

Comment:
Should this be `static` ? Looks like a private variable used just within this class.

Reply:
Can multiple instances of core tools run at same time? It might be better to avoid global variables to store state in that case.

Reply:
Updating from an offlien conversation: we do not expect multiple instances of Core Tools to run simultaneously. The reason the `_requiredNetFrameworkVersion` variable is `static` is to allow it to be used by static methods like `UpdateFrameworkVersions`, which then are easily test-able. I agree in this case it might be better to make these methods just `private`, but it would also require further changes in our tests. I'd be happy to address those changes in a separate PR.